pkgs/applications/office/gnucash/default.nix

```
@@ -26,12 +26,12 @@
 
 stdenv.mkDerivation rec {
   pname = "gnucash";
-  version = "4.9";
+  version = "4.10";
 
   # raw source code doesn't work out of box; fetchFromGitHub not usable
   src = fetchurl {
     url = "https://github.com/Gnucash/gnucash/releases/download/${version}/${pname}-${version}.tar.bz2";
-    hash = "sha256-mlUcMMG3EhmfwiJ6EJr7mE177xjhOBcLvHIlxsH6ty0=";
+    hash = "sha256:0fy9p5fgi2i0x7acg5fnkfdrxxd3dypi3ykvnj53hfbfky8vpm3z";
```

Convert using `nix hash to-sri <sha256:hash>`

```suggestion
    hash = "sha256-f9S7kZ9uOTiKtHv6Ea9vo/Wem5vWlcfU6SCK+Fy5yTs=";
```

---

Why?

I can't find much info about what this changes and why it's necessary/preferred/suggested:

- https://github.com/NixOS/nixpkgs/search?q=sri
- https://nixos.wiki/wiki/Nix_Hash#Usage says this is `supposedly preferred`
- https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-hash-to-sri.html says the command is experimental

---

[pedantry]

> Why?

_Why not?_

If you found the hash in SRI format, why not to keep the format? After all, it was SRI before you change it.

Besides, I already provided you the equivalent SRI-formatted hash, just in the case you are not using the most recent `nix` evaluator. Don't worry, the old Nix tools will recognize the SRI format too.

> 
> I can't find much info about what this changes and why it's necessary/preferred/suggested:
> 

Necessity? This is Nixpkgs, there is nothing necessary as a mathematical truth here. Nonetheless, there is a progressive use of SRI format.
The same way `md5` was deprecated/eliminated in favor of `sha256` from nixpkgs codebase, the same should happen with old format being deprecated by the new SRI.

> * https://github.com/NixOS/nixpkgs/search?q=sri

Code search of github site is less than good. 

On the other hand, the @r-ryantm  bot was updates to use SRI hashes. Besides it, the `fetchurl` source code explicitly says`sha256` is legacy whereas `hash` SRI is preferred:

https://github.com/NixOS/nixpkgs/blob/a08498a0cd8751e4427a2f4729c76eb1cf152e14/pkgs/build-support/fetchurl/default.nix#L54

https://github.com/NixOS/nixpkgs/blob/a08498a0cd8751e4427a2f4729c76eb1cf152e14/pkgs/build-support/fetchurl/default.nix#L57

> * https://nixos.wiki/wiki/Nix_Hash#Usage says this is `supposedly preferred`

It is an argument **for** and not _against_ SRI format. Also, read the next line:

> > If you find a hash that uses colon as a separator (<type>:<hash>), don't use that. This relies on undocumented behavior and is not officially supported.

> * https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-hash-to-sri.html says the command is experimental

`nix` command as a whole is considered experimental, nonetheless it is integrated into the workflow of many people nowadays. It will not be abandoned in the future, indeed it will be integrated and the old `nix-*` will be deprecated/removed.

Indeed, every place you hear or see a `flake.nix` the `nix` command is tacitly being used. The [flake-compat](https://github.com/edolstra/flake-compat) package is precisely the exception that confirms the rule.

Besides, "experimental" is a bad argument for a distro that has an `unstable` channel for distributing fresh packages...

[/pedantry]

---

@StephenWithPH ping